### DISCUSSION TITLE
Cannot setup vercel.json to configure function memory allocation

### DISCUSSION THREAD

**Discussion by LukePrior** [vercel/community#297] [Help] [answered]
Hello everyone,

This is my first time using Vercel and deploying an API so please bear with me if I'm making any obvious mistakes.

I have an API deployed at `https://nbn-service-check.vercel.app/check`

The code for it can be found here: https://github.com/LukePrior/nbn-service-check

I am trying to configure the function to use 256 MB memory instead of 1024 MB so that I don't hit my limit.

I first tried creating a vercel.json file in the root directory with the following:

```
{
    "functions": {
      "api/index.js": {
        "memory": 256
      }
    }
  }
```

This resulted in the following error on deployment: `The pattern "index.js" defined in 'functions' doesn't match any Serverless Functions inside the 'api' directory.`

So I tried moving my index.js file to an API folder and reconfiguring the now.json, package.json, vercel.json files to reference this new location which allowed it to deploy correctly but I could no longer access my API at /check just returning a 404 error.

The code for this setup can be found here: https://github.com/LukePrior/nbn-service-check/tree/test

Does anyone know if this is possible to do?

**Comment by gogocharli** [comment, marked as answer]
Hello @LukePrior, have you been able to configure the functions properly?
I have seen in your repository that the current `vercel.json` file should be working as expected.

**Reply by LukePrior** [reply]
Yes I figured it out eventually.
